oop-customers/Invoice.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: flagoon
+ * Date: 14.01.18
+ * Time: 15:07
+ */
+
+class Invoice
+{
+    protected $invoiceId;
+    protected $sellDate;
+    protected $customer;
+
+    public function setInvoiceID(int $id): void
+    {
+        $this->invoiceId = $id;
+    }
+
+    public function setDate(string $date):void
+    {
+        $this->sellDate = $date;
+    }
+
+    /**
+     * Function that checks if NIP, name and last name is set and if so, it create $customer variable in Invoice.
+     * @param $customer Customer
+     */
+    public function setCustomer(Customer $customer): void
+    {
+        try {
```

Comment:
Widzę to tak, że metoda setCustomer wymusza istnienie wartości wybranych własności obiektu (NIP, imię i nazwisko). Natomiast załatwia to PHP7 poprzez strict type dla zwracanych wartości, więc nie musisz się tym martwić tutaj.

Temat albo do momentu, aż przerobimy wkrótce wyjątki, albo do pogadania na przerwie. ;)